src/libraries/System.Formats.Tar/src/System/Formats/Tar/TarHeader.Write.cs

```diff
@@ -0,0 +1,652 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Text;
+
+namespace System.Formats.Tar
+{
+    // Writes header attributes of a tar archive entry.
+    internal partial struct TarHeader
+    {
+        private const byte SpaceChar = 0x20;
+        private const byte EqualsChar = 0x3d;
+        private const byte NewLineChar = 0xa;
+        private static readonly byte[] s_paxMagic = new byte[] { 0x75, 0x73, 0x74, 0x61, 0x72, 0x0 }; // "ustar\0"
+        private static readonly byte[] s_paxVersion = new byte[] { 0x30, 0x30 }; // "00"
+
+        private static readonly byte[] s_gnuMagic = new byte[] { 0x75, 0x73, 0x74, 0x61, 0x72, 0x20 }; // "ustar "
+        private static readonly byte[] s_gnuVersion = new byte[] { 0x20, 0x0 }; // " \0"
```

Comment:
Can these be spans instead? e.g.
```C#
private static ReadOnlySpan<byte> PaxMagic => new byte[] { ... };
```
?

Reply:
I noticed the suggestion is a property. Is this smart enough to cache the array, instead of returning new `byte[]` every time it is called?

Reply:
I changed this, but I'm still curious. I suspect the answer is that it gets cached.

Reply:
Roslyn has an optimization to refer to the static data segment of the assembly. So no allocation happens here.
See [sharplab](https://sharplab.io/#v2:EYLgxg9gTgpgtADwGwBYA+ABATARgLABQGAzAATakDCpA3oaQ6QA5QCWAbgIYAuM5OSUrE4ATCADsANgE9SwabwDaAXVIBnAPoBzcQFcAajChrWE0gF5S4mAHc5CmCtqkADAiwuANK4QvSAXwBuUgB6ENIAIlIAHRcI+kYWDh4+DAFSACUYUQB5KWkAZSZOcQAeeV4APlIAcT1DY1NxC2rrOwrHVRofD283PyDQ8KjY+IJ/IA===).